Library/Homebrew/cask/artifact/symlinked.rb

```diff
@@ -40,17 +40,25 @@ def summarize_installed
 
       private
 
-      def link(**options)
+      def link(force: false, **options)
         unless source.exist?
           raise CaskError,
                 "It seems the #{self.class.link_type_english_name.downcase} " \
                 "source '#{source}' is not there."
         end
 
         if target.exist?
-          raise CaskError,
-                "It seems there already exists #{self.class.english_article} " \
-                "#{self.class.english_name} at '#{target}'; not overwriting."
+          message = "It seems there is already #{self.class.english_article} " \
+                    "#{self.class.english_name} at '#{target}'"
+
+          raise CaskError, "#{message}." unless force
+
+          raise CaskError, "#{message}." unless target.symlink?
+
+          raise CaskError, "#{message}." unless target.realpath == source.realpath
```

Comment:
I think we still need an additional check for those pointing into the `Caskroom`, since we still want override links when they point to an older cask version in the `Caskroom`.

Reply:
Also, please try refactoring this into an `if`.

Reply:
Initially I had the line:
```
raise CaskError, "#{message}." unless target.readlink.to_s.include? cask.caskroom_path.to_s
```
which would handle symbolic links pointing to the `Caskroom`, but this would fail for symbolic links that point to `/Applications`.

Reply:
> Also, please try refactoring this into an if.

I tried, but `brew style` complained and suggested `unless` (if I recall).

Reply:
Try merging the conditions into a single statement.

Reply:
```ruby
target.realpath == source.realpath || target.realpath.to_s.start_with?("#{cask.caskroom_path}/")
```

Reply:
Thank you (as always) @reitermarkus -- I have pushed some more updates.

The `if` block looks weird to me, but it passes `style` this way. Perhaps there is a nicer way to do it that also satisfies `style`?

Reply:
Try reversing the if, so the `raise` is in else branch and `target.delete` is in the then branch.

Reply:
Done, much cleaner now.